CRUDRecipeTests/Services/SQLiteIngredientServiceTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿using AutoMapper;
+using CRUDRecipeEF.BL.DL.Data;
+using CRUDRecipeEF.BL.DL.DTOs;
+using CRUDRecipeEF.BL.DL.Helpers;
+using CRUDRecipeEF.BL.DL.Services;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace CRUDRecipeTests.Services
+{
+    public class SQLiteIngredientServiceTests : IngredientServiceTests, IDisposable
+    {
+        private MapperConfiguration autoMapperConfig;
+
+        public SQLiteIngredientServiceTests() : 
+            base(new DbContextOptionsBuilder<RecipeContext>().UseSqlite("Filename=TestIng.db")
+                .Options)
+        {
+            autoMapperConfig = new MapperConfiguration(cfg => cfg.AddProfile<AutoMapperProfiles>());
+            // TODO the configuration is not valid
+            //autoMapperConfig.AssertConfigurationIsValid();
+        }
+
+        public void Dispose()
+        {
+            using (var context = new RecipeContext(ContextOptions))
+            {
+                context.Database.EnsureDeleted();
+            }
+        }
+
+        [Fact]
+        public async Task Test_GetIngredientByName()
+        {
+            using (var context = new RecipeContext(ContextOptions))
+            {
+                var ingredientService = new IngredientService(context, new Mapper(autoMapperConfig));
+                var ingredient = await ingredientService.GetIngredientByName("Apple");
+    
+                Assert.NotNull(ingredient);
+                Assert.Equal("Apple", ingredient.Name);
+            }
+        }
+
+        [Fact]
+        public async Task Test_GetAllIngredients()
+        {
+            using (var context = new RecipeContext(ContextOptions))
+            {
+                var ingredientService = new IngredientService(context, new Mapper(autoMapperConfig));
```

Comment:
Instead of making a new mapper, can we create it in the ctor and reuse the field?

Reply:
On another branch now, will take a look in a moment, but probably

Reply:
Done, good to merge now :D